qemu/tests/virtio_mem_hugetlb_migration.py

```diff
@@ -0,0 +1,58 @@
+import time
+
+from avocado.utils import process
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Do migration of a virtio-mem device consuming hugepages
+    1) Setup hugepages in source and destination hosts
+    2) Boot VMs in source and destination hosts
+    3) Do migration
+    4) Check virtio-mem device
+    5) Resize virtio-mem devices to a different value
+    6) Check virtio-mem device
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    vm.wait_for_login()
+    threshold = params.get_numeric("threshold", target_type=float)
+    mem_object_id = params["mem_devs"].split().pop()
+    mem = params.get_numeric("mem")
+    initial_req_size = params.get("requested-size_memory_%s" % mem_object_id)
+    requested_size_vmem = params.get("requested-size_test_%s" % mem_object_id)
+    device_id = "virtio_mem-%s" % mem_object_id
+    node_id = params.get_numeric("node_memory_%s" % mem_object_id)
+    process.system("echo %d > /proc/sys/vm/nr_hugepages" % mem, shell=True)
```

Comment:
With `setup_hugepage=yes` in cfg,  env_process will setup hugepage based on some logic, checking hugepage size and setup the target hugepages.
Logically, i think it is conflict with the setting in the test case level.

And,  `nr_hugepages` is the page number, i did not see any checking about the page size.

Reply:
@yanan-fu 
I get that using `nr_hugepages` doesn't keep in mind the hugepages size, so likely that needs to be changed, but then how could be handled the self migraton of 2 VMs using hugepages? AFAIU `setup_hugepage` only does it for the source VM, right? I need the double of those hugepages...

Reply:
Actually,  `setup_hugepage` is about the action in host env, both src and dst vm can use the hugepage path.

Reply:
@yanan-fu could you review again this PR? Thanks !